### DISCUSSION TITLE
Linter rule for unused imports?

### DISCUSSION THREAD

**Discussion by spg** [rome/tools#4188] [Suggestions]
Is there a rule that allows detection of unused imports?

**Comment by Conaclos** [comment]
For now, there is no rule that covers it.

ESLint reports unused imports when `no-unused-var` is enabled.
This behavior was not implemented in `noUnusedVariables`.
I do not know if this was intentional.

We should either add this check on  `noUnusedVariables` or create a dedicated rule  `noUnusedImports`.

**Comment by jpike88** [comment]
It's also detected by typescript, violating `ts(6133)`.

**Reply by neurosnap** [reply]
I wonder if that's why they didn't bother implementing it?

**Reply by zachequi** [reply]
TSC wouldn't auto-fix the rule, and even if it did, it's way way slower than rome.  I'd double vote for rome implementing this rule.

**Comment by arendjr** [comment]
Looks like the `noUnusedVariables` rule in `main` covers this, see: https://github.com/rome/tools/issues/4685
